src/NCronJob/Registry/JobDefinition.cs

```diff
@@ -1,24 +1,52 @@
 using Cronos;
+using System.Diagnostics.CodeAnalysis;
 
 namespace NCronJob;
 
-internal sealed record JobDefinition(
-    Type Type,
-    object? Parameter,
-    CronExpression? CronExpression,
-    TimeZoneInfo? TimeZone,
-    string? JobName = null,
-    JobExecutionAttributes? JobPolicyMetadata = null)
+internal sealed record JobDefinition
 {
+    private JobDefinition(
+        Type type,
+        object? parameter)
+    {
+        if (type.FullName is null // FullName is later required to properly identify the JobDefinition
```

Comment:
As you introduced static construction methods, wouldn't it be less surprising to have the exception in there rather than throwing from the ctor?

Reply:
Fixed